Utils/Dataset_formatter_xsiem/dataset_formatter_xsiem.py

```diff
@@ -0,0 +1,239 @@
+
+    @staticmethod
+    def replace_day_with_number(date: str):
+        """
+        Args:
+            - a date of type Aug 3rd 2022 18:40:17
+
+        Returns:
+            - The same date witout the (st|nd|rd|th)
+        """
+        return re.sub(r'(st|nd|rd|th)', r'', date)
+
+
+class TimeFormat2(Formatter):
+
+    @property
+    def name(self):
+        return 'TimeFormat2 example - 2022-08-03T15:40:16.986Z'
+
+    @property
+    def regex(self):
+        return r"\d{4}(-\d{2}){2}T(\d{2}:){2}\d{2}.\d{3}Z"
+
+    def convert_to_xsiam_format(self, col):
+        res_list = []
+        for time_element in col:
+            if isinstance(time_element, float):
+                res_list.append('2003-01-01T00:00:00.000Z')
+            else:
+                res_list.append(time_element)
+
+        return res_list
+
+
+class JsonFormat(Formatter):
+
+    @property
+    def name(self):
+        return 'Json Format - {example}'
+
+    @property
+    def regex(self):
+        return r"^{.*}$"
+
+    def convert_to_xsiam_format(self, col):
+        res_list = []
+        for cell in col:
+            if isinstance(cell, float) and math.isnan(cell):
+                res_list.append('{}')
+            else:
+                cell = cell.replace('\\', '\\\\')
+                res_list.append(cell)
+
+        return res_list
+
+
+class JsonArrayFormat(Formatter):
+
+    @property
+    def name(self):
+        return 'Json Array Format - []'
+
+    @property
+    def regex(self):
+        return r"^\[.*\]$"
+
+    def convert_to_xsiam_format(self, col):
+        res_list = []
+        for cell in col:
+            if isinstance(cell, float) and math.isnan(cell):
+                res_list.append('[]')
+            else:
+                cell = cell.replace('\\', '\\\\')
+                res_list.append(cell)
+
+        return res_list
+
+
+class FormatManager:
+    def __init__(self):
+        self.format_type: Formatter = None
+        self.formatters: list[Formatter] = [TimeFormat1(), JsonFormat(), TimeFormat2(), JsonArrayFormat()]
+
+    def find_col_format(self, col: list):
+        """
+        If a format was detected it puts that format class in self.format_type
+        else it sets the self.format_type to None
+
+        Args:
+            - One of the dataFrame cols
+
+        Returns:
+            - None
+        """
+        for val in col:
+            for formatter in self.formatters:
+                if formatter.is_format(val):
+                    self.format_type = formatter
+                    return 
+        
+        self.format_type = None
+
+    def convert_to_xsiam_format(self, col):
+        """
+        Args:
+            - One of the dataFrame cols
+
+        Return:
+            - A list of the formatted col
+        """
+        return self.format_type.convert_to_xsiam_format(col)
+
+    def formatter_name(self):
+        return self.format_type.name
+
+
+def open_input_file(path: Path, f):
+    if path.suffix == '.tsv':
```

Comment:
Do you support only tsv files ? If so please update the description in the parser

Reply:
No i support also csv files. 
This is specified in the script parameters.
'A Script to format csv/tsv files in order to upload to XSIAM'
here in case the file is a tsv format we read it with a sep='\t' flag.